### PR TITLE
[freeboxos] FreeboxOsIconProvider should only provide icons for its own icon set

### PR DESCRIPTION
**Pull request by clinique**
Currently the FreeboxOsIconProvider tries to provide icons for every icon requests. This should be restricted to `freeboxos:iconname`

Updated ipaddress dependency because previous version was marked with one known vulnerability.

### CODE REVIEW

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java [hunk 1]
```diff
@@ -50,25 +55,41 @@
...
+        String label = getText("label", DEFAULT_LABEL, locale);
+        String description = getText("decription", DEFAULT_DESCRIPTION, locale);
+
+        return Set.of(new IconSet(BINDING_ID, label, description, Set.of(Format.SVG)));
+    }
+
+    private String getText(String entry, String defaultValue, @Nullable Locale locale) {
+        String text = defaultValue;
+        if (locale != null) {
+            text = i18nProvider.getText(context.getBundle(), "iconset." + entry, defaultValue, locale);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java:89]: I believe you can directly call `i18nProvider.getText` without checking first if locale is null.

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java [hunk 2]
```diff
@@ -50,25 +55,41 @@
 @NonNullByDefault
 @Component(immediate = true, service = { IconProvider.class })
 public class FreeboxOsIconProvider extends AbstractResourceIconProvider {
+    private static final String DEFAULT_DESCRIPTION = "Icons provided by FreeboxOS iteself";
```
[the remark this answers is not in the document]
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java:58]: Corrected

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java (hunk 1 above)
Comment (clinique) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java:89]: You're right.

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java [hunk 3]
```diff
@@ -50,25 +55,38 @@
...
                 .path("resources/images/home/pictos");
     }
 
     @Override
     public Set<IconSet> getIconSets(@Nullable Locale locale) {
-        return Set.of();
+        String label = getText("label", DEFAULT_LABEL, locale);
+        String description = getText("decription", DEFAULT_DESCRIPTION, locale);
+
+        return Set.of(new IconSet(BINDING_ID, label, description, Set.of(Format.SVG)));
```
Comment (lolodomo) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/FreeboxOsIconProvider.java:84]: Icons are apparently PNG and not SVG if I consider your method hasResource.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**